src/cpu/cpu_engine.hpp

```diff
@@ -37,7 +37,14 @@
     impl_list_item_t( \
             impl_list_item_t::type_deduction_helper_t<__VA_ARGS__::pd_t>())
 #define CPU_INSTANCE(...) DNNL_PRIMITIVE_IMPL(CPU_INSTANCE_IMPL, __VA_ARGS__)
+#ifdef _WIN32
+// Expanding DNNL_X64_ONLY for Windows to fix compilation failure on Windows + CPU plugin.
+// DNNL_X64_ONLY == CONCAT2(Z_DO_IF_, DNNL_X64)
+#define CPU_INSTANCE_X64(...) \
+    CONCAT2(Z_DO_IF_, DNNL_X64)(CPU_INSTANCE(__VA_ARGS__))
+#else
 #define CPU_INSTANCE_X64(...) DNNL_X64_ONLY(CPU_INSTANCE(__VA_ARGS__))
+#endif
```

Comment:
Have you tried to replace it entirely with the CONCAT2 version both for Linux and Windows?

Reply:
Yes, it works for Linux and Windows.

Reply:
So maybe keep a single version of it then? It also seems a candidate for oneDNN PR promotion.

Reply:
Let's wait CI: win-cc test result.

Reply:
@dzarukin I saw all CI test pass.  Just update: keeping algin between Windows and Linux look like better.